File: Backend/bdu-clearance/src/main/java/com/bdu/clearance/services/StudentService.java
package com.bdu.clearance.services;

import com.bdu.clearance.dto.student.StudentRequestDto;
import com.bdu.clearance.dto.student.StudentResponseDto;
import com.bdu.clearance.enums.StudentStatus;

import java.util.List;

public interface StudentService {

    List<StudentResponseDto> getAllStudents();

    void createStudent(StudentRequestDto studentDto);

    void deleteStudent(String studentId);

    void updateStudent(StudentRequestDto studentDto, Long id);

    StudentResponseDto getStudent(String studentId);

    List<StudentResponseDto> getStudentByAdvisor(Long advisorId);

    List<StudentResponseDto> getStudentByStatus(StudentStatus status);

    List<StudentResponseDto> getStudentByYearOfStudy(int year);
}

File: Frontend/bdu-clearance/src/services/studentService.js
import api from "./api";

const BASE_URL = "/public/students";

const createStudent = async (data) => {
  await api.post(`${BASE_URL}/create`, data);
};

const getAllStudents = async () => {
  const response = await api.get(BASE_URL);
  return response.data;
};

const getStudent = async (studentId) => {
  const response = await api.get(`${BASE_URL}/${studentId}`);
  console.log("studentId", studentId);
  console.log(response.data);
  return response.data;
};

const updateStudent = async (id, data) => {
  await api.put(`${BASE_URL}/${id}`, data);
};

const deleteStudent = async (studentId) => {
  await api.delete(`${BASE_URL}/${studentId}`);
};

const studentService = {
  createStudent,
  getAllStudents,
  getStudent,
  updateStudent,
  deleteStudent,
};

export default studentService;
